File: app/src/main/java/abandonedstudio/app/cookingapp/Database/DishCategory.java
package abandonedstudio.app.cookingapp.Database;

import androidx.room.ColumnInfo;
import androidx.room.Entity;
import androidx.room.Index;
import androidx.room.PrimaryKey;

@Entity(tableName = "dish_category", indices = {@Index("categoryId")})
public class DishCategory {

    public DishCategory(String category) {
        this.category = category;
    }

    @PrimaryKey(autoGenerate = true)
    private int categoryId;

    @ColumnInfo(name = "category")
    private String category;

    public int getCategoryId() {
        return categoryId;
    }

    public void setCategoryId(int categoryId) {
        this.categoryId = categoryId;
    }

    public String getCategory() {
        return category;
    }
}

File: app/src/main/java/abandonedstudio/app/cookingapp/core/data/local/room/dish_category/DishCategory.kt
package abandonedstudio.app.cookingapp.core.data.local.room.dish_category

import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.Index
import androidx.room.PrimaryKey

@Entity(tableName = "dish_category", indices = [Index("categoryId")])
data class DishCategory(

    @ColumnInfo(name = "category")
    var category: String,

    @PrimaryKey(autoGenerate = true)
    var categoryId: Int? = null

)
